modules/personalize_taxonomy_context/personalize_taxonomy_context.module

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * @file personalize_taxonomy_context.module
+ * Provides a visitor context plugin for taxonomy.
+ */
+
+/**
+ * Implements hook_personalize_visitor_contexts().
+ */
+function personalize_taxonomy_context_personalize_visitor_context() {
+  $info = array();
+  $path = drupal_get_path('module', 'personalize_taxonomy_context') . '/plugins';
+  $info['taxonomy_context'] = array(
+    'path' => $path . '/visitor_context',
+    'handler' => array(
+      'file' => 'TaxonomyContext.inc',
+      'class' => 'TaxonomyContext',
+    ),
+  );
+  return $info;
+}
+
+/**
+ * This function is expected to exist and be used to retrieve terms.
+ *
+ * @param $nid
+ *   Node id.
+ * @param $vid
+ *   Vocabulary id.
+ * @return array
+ *   An array of string terms.
+ */
+function personalize_taxonomy_context_get_terms($nid, $vid) {
+  $query =
+    'SELECT {taxonomy_term_data}.name ' .
+    'FROM {taxonomy_term_data} ' .
+    'JOIN {taxonomy_vocabulary} ' .
+    'ON {taxonomy_term_data}.vid = {taxonomy_vocabulary}.vid ' .
+    'JOIN {taxonomy_index} ' .
+    'ON {taxonomy_term_data}.tid = {taxonomy_index.tid} ' .
+    'WHERE {taxonomy_index}.nid = :nid ' .
+    'AND {taxonomy_vocabulary}.vid = :vid';
+
```

Comment:
So I see Personalize module already requires Visitor Actions module, and Visitor Actions already does this: https://github.com/acquia/visitor_actions/blob/7.x-1.x/visitor_actions.module#L624

This means my personalize_taxonomy_context.js can already find the terms in Drupal.settings.visitor_actions.pageContext.Taxonomy. Is there any reason why we should / can not use it?

One catch here is that to do so I may need to page_build more value into to Drupal.settings.visitor_actions.pageContext.Taxonomy, just because taxonomy context needs them.